transmit-lib/src/cmr/transmit/kms.clj

```diff
@@ -121,6 +130,34 @@
                        (pr-str expected-subfield-names)
                        (pr-str subfield-names)))))))
 
+(defn- parse-entries-from-xml
+  "Parses the XML returned by the GCMD KMS. Only when doing the idnnode validation, the kms file
+   is in xml format - concept.xml. The first 4 lines are headers. The rest of the lines are actual
+   data. Those lines with conceptScheme=\"idnnode\", the values of prefLabel are the valid ShortNames.
+   for the DirectoryName:
+   <conceptBrief id=\"296143\" uuid=\"03d77986-98aa-4ad4-9070-2b7a41eadb31\" prefLabel=\"SLOAN\" conceptSchemeId=\"599\" conceptScheme=\"idnnode\" isLeaf=\"true\" status=\"\"/> 
+   Note: even though it's an xml file, we could still use read-csv to read it because they are just lines of strings.
+   Returns a sequence of full hierarchy maps."
+  [keyword-scheme xml-content]
+  (let [all-lines (csv/read-csv xml-content)
```

Comment:
In the xpath code, you can actually filter so you'd be able to filter where conceptScheme=idnnode. We do this a lot for iso parsing. So you could do `(select element xpath)` where xpath is the path with the filter on it. Then you would not have to use regular expressions. It would cut down the code a bit. I can help with specific code to do this.